### PR TITLE
Updated files to 4.3.1

### PR DESCRIPTION
**Pull request by RobinAtherton**
- added hello world interceptor
- updated licenses
- updated dependencies

### CODE REVIEW

File: src/main/resources/META-INF/services/com.hivemq.extension.sdk.api.ExtensionMain [hunk 1]
```diff
@@ -1 +1 @@
-com.hivemq.extensions.HelloWorldMain
\ No newline at end of file
+com.hivemq.HelloWorldMain
```
Comment (YannickWeber) [inline src/main/resources/META-INF/services/com.hivemq.extension.sdk.api.ExtensionMain:1]: revert

File: README.adoc [hunk 2]
```diff
@@ -1,46 +1,58 @@
 :hivemq-link: http://www.hivemq.com
 :hivemq-extension-docs-link: http://www.hivemq.com/docs/extensions/latest/
-:hivemq-extension-docs-archetype-link: https://www.hivemq.com/docs/4/extensions/getting-started.html#archetype
+:hivemq-extension-docs-archetype-link: http://www.hivemq.com/docs/extensions/latest/#maven-archetype-chapter
```
Comment (SgtSilvio) [inline README.adoc:3]:
```suggestion
:hivemq-extension-docs-archetype-link: https://www.hivemq.com/docs/extensions/latest/#maven-archetype-chapter
```

File: src/main/resources/hivemq-extension.xml [hunk 3]
```diff
@@ -17,8 +17,8 @@
 
 <hivemq-extension>
     <id>${artifactId}</id>
-    <name>${extension.name}</name>
     <version>${version}</version>
+    <name>${extension.name}</name>
```
Comment (SgtSilvio) [inline src/main/resources/hivemq-extension.xml:21]: why was this moved?

Comment (YannickWeber) [reply src/main/resources/hivemq-extension.xml:21]: to unify it with the maven archetype
